hudi-common/src/main/java/org/apache/hudi/metadata/BaseTableMetadata.java

```diff
@@ -65,14 +67,15 @@
 
   // Directory used for Spillable Map when merging records
   protected final String spillableMapDirectory;
-  private transient HoodieMetadataMergedInstantRecordScanner timelineRecordScanner;
+  private TimelineMergedTableMetadata timelineMergedMetadata;
```

Comment:
this is also serialized now, so we don't list the data timeline for merging over and over.